### PR TITLE
scripts: migrate non-essential scripts to separate repo

### PR DESCRIPTION
**Pull request by knz**
Suggested by @bdarnell in https://github.com/cockroachdb/cockroach/pull/17988#issuecomment-336678825:

The script `pprof` is retired, as its documentation indicated it is
becoming obsolete as of Go 1.9 which is now current.

The script `travis_retry.sh` is retired, as we are not using Travis
any more.

The script `verify-release-binaries.sh` is retired, as it was replaced
by http://github.com/benesch/bincheck/.

The following scripts have been migrated, together with their history,
to the new repository at https://github.com/cockroachdb/scripts:

- `authors.sh`
- `bench`
- `crlog.vim`
- `crlogictest.vim`
- `git-blame-stats.sh`
- `log2html.sh`
- `purge-terraform.sh`
- `torch`

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -342,11 +342,11 @@ $(ARCHIVE): $(ARCHIVE).tmp
 	gzip -c $< > $@
 
 # TODO(benesch): Make this recipe use `git ls-files --recurse-submodules`
-# instead of scripts/ls-files.sh once Git v2.11 is widely deployed.
+# instead of build/ls-files.sh once Git v2.11 is widely deployed.
 .INTERMEDIATE: $(ARCHIVE).tmp
 $(ARCHIVE).tmp: ARCHIVE_BASE = cockroach-$(shell cat .buildinfo/tag)
 $(ARCHIVE).tmp: .buildinfo/tag .buildinfo/rev .buildinfo/basebranch
-	scripts/ls-files.sh | $(TAR) -cf $@ -T - $(TAR_XFORM_FLAG),^,$(ARCHIVE_BASE)/src/github.com/cockroachdb/cockroach/, $^
+	build/ls-files.sh | $(TAR) -cf $@ -T - $(TAR_XFORM_FLAG),^,$(ARCHIVE_BASE)/src/github.com/cockroachdb/cockroach/, $^
```
Comment (benesch) [inline Makefile]: Thanks!

File: scripts/travis_retry.sh [hunk 2]
```diff
@@ -1,49 +0,0 @@
-#!/usr/bin/env bash
```
Comment (benesch) [inline scripts/travis_retry.sh]: This script is used by azworker and gceworker

**Review by benesch — COMMENTED** [review]
Thanks for taking this on, @knz! I'd personally prefer if we kept

* crlog.vim
* crlogictest.vim
* log2html.sh
* purge-terraform.sh

as those are scripts that are tightly coupled to this repository. (Though on the last one I'm particularly biased.) The others are scripts that are generally useful, interact mostly with Git/Go, so might as well live elsewhere.

**Comment by benesch** [comment]
Probably wait for @bdarnell's opinion before making the changes I suggested.

**Comment by bdarnell** [comment]
We definitely need to keep travis_retry.sh since it's used by the `*worker` scripts. (it's cargo-culted from somewhere and should probably be renamed; we never used travis in this repo). The git authors/blame scripts definitely belong in a separate repo. I'm more ambivalent about the log formatters and `bench`, which are more cockroach-specific. If we're going to have a separate repo I'd probably move them there, though. 

Unfortunately, we still need `scripts/pprof`. The `https+insecure` mode still doesn't work in Go 1.9. It's not cockroach-specific, though, so I guess for the sake of consistency it belongs in the other repo.

**Comment by bdarnell** [comment]
On a more general note, I'm not 100% on board with the idea of this scripts repo - my issue with these git-parsing scripts is not that they should live in a cockroach-owned location but not in the main repo, it's that I think they don't really belong in a cockroach-owned location at all. They're more personal - are any of these scripts used by more than one person?

**Comment by dt** [comment]
I'm 👎 on separate repos (in general) -- if/when we want to move, reconfigure, etc things, it is just more that has to move, *manually*, in lockstep.

**Comment by tschottdorf** [comment]
I'm fine nuking all of the (non-essential) scripts and having their owners host them in their own repos. The idea with this "scripts" folder was that things would ultimately be useful, but I'm not sure that they are.

**Comment by dt** [comment]
strawman: we could make a `contrib` folder (either top level with a `scripts` folder, or just under scripts) for "someone finds this useful, so here it is, ymmv" stuff, potentially including some of these scripts which aren't referenced in any of our guides/docs/workflows.

**Comment by jordanlewis** [comment]
FWIW, I think it's super convenient to have scripts like `scripts/bench` accessible from the cockroach working directory. I use that one all the time.

**Pull request closed without merging**